test/spec/modules/lotamePanoramaIdSystem_spec.js

```diff
@@ -440,10 +440,153 @@ describe('LotameId', function() {
+        'https://id.crwdcntrl.net/id?gdpr_applies=true&gdpr_consent=consentGiven'
+      );
+    });
+  });
+
+  describe('when gdpr applies and falls back to euconsent cookie', function () {
+    let request;
+    let callBackSpy = sinon.spy();
+    let consentData = {
+      gdprApplies: true,
+      consentString: undefined
+    };
+
+    beforeEach(function () {
+      getCookieStub
+        .withArgs('euconsent-v2')
+        .returns('consentGiven');
+
+      let submoduleCallback = lotamePanoramaIdSubmodule.getId({}, consentData).callback;
+      submoduleCallback(callBackSpy);
+
+      // the contents of the response don't matter for this
+      request = server.requests[0];
+      request.respond(200, responseHeader, '');
+    });
+
+    it('should call the remote server when getId is called', function () {
+      expect(callBackSpy.calledOnce).to.be.true;
+    });
+
+    it('should pass the gdpr consent string back', function() {
+      expect(request.url).to.be.eq(
+        'https://id.crwdcntrl.net/id?gdpr_applies=true&gdpr_consent=consentGiven'
+      );
+    });
+  });
+
+  describe('when gdpr applies but no consent string is available', function () {
+    let request;
+    let callBackSpy = sinon.spy();
+    let consentData = {
+      gdprApplies: true,
+      consentString: undefined
+    };
+
+    beforeEach(function () {
+      let submoduleCallback = lotamePanoramaIdSubmodule.getId({}, consentData).callback;
+      submoduleCallback(callBackSpy);
+
+      // the contents of the response don't matter for this
+      request = server.requests[0];
+      request.respond(200, responseHeader, '');
+    });
+
+    it('should call the remote server when getId is called', function () {
+      expect(callBackSpy.calledOnce).to.be.true;
+    });
+
+    it('should not include the gdpr consent string on the url', function() {
+      expect(request.url).to.be.eq(
+        'https://id.crwdcntrl.net/id?gdpr_applies=true'
+      );
+    });
+  });
+
+  describe('when no consentData and falls back to eupubconsent cookie', function () {
+    let request;
+    let callBackSpy = sinon.spy();
+    let consentData;
+
+    beforeEach(function () {
+      getCookieStub
+        .withArgs('eupubconsent-v2')
+        .returns('consentGiven');
+
+      let submoduleCallback = lotamePanoramaIdSubmodule.getId({}, consentData).callback;
+      submoduleCallback(callBackSpy);
+
+      // the contents of the response don't matter for this
+      request = server.requests[0];
+      request.respond(200, responseHeader, '');
+    });
+
+    it('should call the remote server when getId is called', function () {
+      expect(callBackSpy.calledOnce).to.be.true;
+    });
+
+    it('should pass the gdpr consent string back', function() {
+      expect(request.url).to.be.eq(
+        'https://id.crwdcntrl.net/id?gdpr_consent=consentGiven'
+      );
+    });
+  });
+
+  describe('when no consentData and falls back to euconsent cookie', function () {
```

Comment:
maybe one more - when no consent data and no fallback cookies?

Reply:
Sure thing!